remoulade/api/main.py

```diff
@@ -1,30 +1,36 @@
 """ This file describe the API to get the state of messages """
+from operator import itemgetter
+
 from flask import Flask, request
 from marshmallow import ValidationError
-from werkzeug.exceptions import BadRequest, HTTPException, NotFound
+from werkzeug.exceptions import HTTPException, NotFound
 
 import remoulade
 from remoulade import get_broker, get_scheduler
 from remoulade.errors import NoScheduler, RemouladeError
-from remoulade.state import StateNamesEnum
 
-from .schema import MessageSchema
+from .schema import MessageSchema, PageSchema
 
 app = Flask(__name__)
 
-
+# TODO: return the number of register, to compute the number of pages
 @app.route("/messages/states")
 def get_states():
-    name_state = request.args.get("name")
+    payload = PageSchema().load(request.args.to_dict())
     backend = remoulade.get_broker().get_state_backend()
     data = [s.as_dict() for s in backend.get_states()]
-    if name_state is not None:
-        name_state = name_state.lower().capitalize()
-        try:
-            state = StateNamesEnum(name_state)
-            data = [s for s in data if s["name"] == state.value]
-        except ValueError:
-            raise BadRequest("{} state is not defined.".format(name_state))
+    if "search_keys" in payload:
+        keys = payload["search_keys"].split(",")
+        value = payload["search_value"].lower()
+        data = [item for item in data if "~".join([str(item[k]) for k in keys]).lower().find(value) >= 0]
```

Comment:
maybe use a more special character than this (like ◬), else a user could  type the character and not filter anything.